File: src/main/java/upbrella/be/rent/dto/response/RentFormResponse.java
package upbrella.be.rent.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.umbrella.entity.Umbrella;

@Getter
@Builder
public class RentFormResponse {

    private String classificationName;
    private long storeMetaId;
    private String rentStoreName;
    private long umbrellaUuid;

    public static RentFormResponse of(Umbrella umbrella) {

        return RentFormResponse.builder()
                .classificationName(umbrella.getStoreMeta().getClassification().getName())
                .storeMetaId(umbrella.getStoreMeta().getId())
                .rentStoreName(umbrella.getStoreMeta().getName())
                .umbrellaUuid(umbrella.getUuid())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/rent/dto/response/RentFormResponse.kt
package upbrella.be.rent.dto.response

import upbrella.be.umbrella.entity.Umbrella

data class RentFormResponse(
    val classificationName: String,
    val storeMetaId: Long,
    val rentStoreName: String,
    val umbrellaUuid: Long,
    val hasPhoneNumber: Boolean
) {
    companion object {
        fun of(umbrella: Umbrella, hasPhoneNumber: Boolean): RentFormResponse {
            return RentFormResponse(
                classificationName = umbrella.storeMeta.classification!!.name!!,
                storeMetaId = umbrella.storeMeta.id!!,
                rentStoreName = umbrella.storeMeta.name,
                umbrellaUuid = umbrella.uuid,
                hasPhoneNumber = hasPhoneNumber
            )
        }
    }
}
